### ISSUE TITLE
[BUG]: Go gc doesn't display x86 assembly

### ISSUE THREAD

**Issue by siboehm** [compiler-explorer/compiler-explorer#4735]
### Describe the bug

There seems to be no way to look at x86 assembly output of the Go gc compiler. By default CE displays Go assembly (which is fine), but I want to look at x86 asm.

### Steps to reproduce

1. Input Go code: https://go.godbolt.org/z/z1W1P8zsc
2. Click link to binary
3. "No assembly to display"

### Expected behavior

Displays x86 assembly

### Reproduction link

https://go.godbolt.org/z/z1W1P8zsc

### Screenshots

Not applicable

**Comment by dkm** [comment]
Hmmm, so this has never worked, I think. Looks like we need some tuning to have both Go assembly in the first case, and use target-specific assembly in the linker/binary object case. Currently, we don't call objdump or try to use the correct AsmParser (from a quick first look at our go support)

**Comment by siboehm** [comment]
It's possible that I might be misremembering regarding this having worked in the past. I had assumed that "Link to binary" was supposed to do `Go --compile-with-gc--> static binary ---objump---> x86 `.

**Comment by dkm** [comment]
Yes, that's what it should do, but currently, it's not. Maybe I need to take another look at the code, maybe it was working and there's a simple fix.
